pa11y/webapp/write-report.js

```diff
@@ -63,8 +74,39 @@ const promises = urls.map(url =>
 
 Promise.all(promises)
   .then(async results => {
-    await fs.promises.mkdir('./.dist', { recursive: true });
-    await writeFile('./.dist/report.json', JSON.stringify({ results }));
-    console.info('Reporting done!');
+    if (isPullRequestRun) {
```

Comment:
Do we need to vary what we do here? Can this fail without issue on a non-PR run?

Reply:
Again not sure what's meant here 🙈 [This has failed in the past on non-PR runs](https://github.com/wellcomecollection/wellcomecollection.org/actions/runs/10242411617) (as it runs in prod) and [we now have a mechanism that sends us a Slack message should this happen](https://github.com/wellcomecollection/wellcomecollection.org/pull/11093).

Reply:
This `if` statement either:
- writes a report and exits successfully
- logs the contents of a report and exits 0 or 1 based on if there are any errors

I'm wondering if we really need to do different things? In both cases could you:

- write a report
- log the contents of the report
- exits 0 or 1 based on if there are any errors

Removing the need to know if running in a PR context.

Reply:
I don't think we don't want it to fail on the daily run, the daily run's job is to update the dashboard. To me, if that succeeds, it shouldn't mark the task as failed because it's found things to fix?

I think it used to do that when it was part of deployments, and it was a blocker to the deployment.

Reply:
[#11213](https://github.com/wellcomecollection/wellcomecollection.org/issues/11213) mentioned here